File: python/docmost/services/mfa_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import DisableMfaDto, EnableMfaDto, MfaDto, RegenerateBackupCodesDto


class MFAApi:
    """MFA API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def mfa_setup(self) -> ApiResponse[Any]:
        return self._http.post("mfa/setup", body=None)

    def mfa_enable(self, request: EnableMfaDto) -> ApiResponse[Any]:
        return self._http.post("mfa/enable", body=request.model_dump(by_alias=True, exclude_none=True))

    def mfa_disable(self, request: DisableMfaDto) -> ApiResponse[Any]:
        return self._http.post("mfa/disable", body=request.model_dump(by_alias=True, exclude_none=True))

    def mfa_status(self) -> ApiResponse[Any]:
        return self._http.post("mfa/status", body=None)

    def mfa_regenerate_backup_codes(self, request: RegenerateBackupCodesDto) -> ApiResponse[Any]:
        return self._http.post("mfa/generate-backup-codes", body=request.model_dump(by_alias=True, exclude_none=True))

    def mfa_verify(self, request: MfaDto) -> ApiResponse[Any]:
        return self._http.post("mfa/verify", body=request.model_dump(by_alias=True, exclude_none=True))

    def mfa_validate_access(self) -> ApiResponse[Any]:
        return self._http.post("mfa/validate-access", body=None)

File: go/docmost/services/mfa_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// MFAAPI implements MFA API.
type MFAAPI struct {
	http *httpclient.HTTPClient
}

func NewMFAAPI(http *httpclient.HTTPClient) *MFAAPI {
	return &MFAAPI{http: http}
}

func (a *MFAAPI) MfaSetup(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/setup", nil)
}

func (a *MFAAPI) MfaEnable(ctx context.Context, request *models.EnableMfaDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/enable", request)
}

func (a *MFAAPI) MfaDisable(ctx context.Context, request *models.DisableMfaDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/disable", request)
}

func (a *MFAAPI) MfaStatus(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/status", nil)
}

func (a *MFAAPI) MfaRegenerateBackupCodes(ctx context.Context, request *models.RegenerateBackupCodesDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/generate-backup-codes", request)
}

func (a *MFAAPI) MfaVerify(ctx context.Context, request *models.MfaDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/verify", request)
}

func (a *MFAAPI) MfaValidateAccess(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "mfa/validate-access", nil)
}

